File: class_book.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

struct Student {
    string name;
    int attendance;
};

class AttendanceManager {
private:
    vector<Student> students;

public:
    void addStudent(const string& name) {
        students.push_back({name, 0});
    }

    void markAttendance(const string& name) {
        for (auto& student : students) {
            if (student.name == name) {
                student.attendance++;
                cout << "Docházka zaznamenána pro: " << name << endl;
                return;
            }
        }
        cout << "Student nenalezen!" << endl;
    }

    void displayAttendance() {
        cout << "\nPřehled docházky:\n";
        for (const auto& student : students) {
            cout << student.name << " - " << student.attendance << " dní\n";
        }
    }
};

int main() {
    AttendanceManager manager;
    int choice;
    string name;

    do {
        cout << "\n1. Přidat studenta\n2. Zaznamenat docházku\n3. Zobrazit docházku\n4. Konec\nVyberte možnost: ";
        cin >> choice;
        cin.ignore();

        switch (choice) {
            case 1:
                cout << "Zadejte jméno studenta: ";
                getline(cin, name);
                manager.addStudent(name);
                break;
            case 2:
                cout << "Zadejte jméno studenta: ";
                getline(cin, name);
                manager.markAttendance(name);
                break;
            case 3:
                manager.displayAttendance();
                break;
            case 4:
                cout << "Ukončuji aplikaci...\n";
                break;
            default:
                cout << "Neplatná volba, zkuste to znovu.\n";
        }
    } while (choice != 4);

    return 0;
}

File: class_book.java
import java.util.*;

class Student {
    private String jmeno;
    private List<Boolean> dochazka;

    public Student(String jmeno) {
        this.jmeno = jmeno;
        this.dochazka = new ArrayList<>();
    }

    public void zaznamDochazky(boolean pritomen) {
        dochazka.add(pritomen);
    }

    public String getJmeno() {
        return jmeno;
    }

    public List<Boolean> getDochazka() {
        return dochazka;
    }
}

public class DochazkaApp {
    private static List<Student> studenti = new ArrayList<>();
    private static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        while (true) {
            System.out.println("\n1. Přidat studenta");
            System.out.println("2. Zaznamenat docházku");
            System.out.println("3. Zobrazit docházku");
            System.out.println("4. Ukončit");

            System.out.print("Vyberte akci: ");
            int volba = scanner.nextInt();
            scanner.nextLine(); 

            switch (volba) {
                case 1:
                    pridatStudenta();
                    break;
                case 2:
                    zaznamenatDochazku();
                    break;
                case 3:
                    zobrazitDochazku();
                    break;
                case 4:
                    System.out.println("Ukončování programu...");
                    return;
                default:
                    System.out.println("Neplatná volba, zkuste to znovu.");
            }
        }
    }

    private static void pridatStudenta() {
        System.out.print("Zadejte jméno studenta: ");
        String jmeno = scanner.nextLine();
        studenti.add(new Student(jmeno));
        System.out.println("Student " + jmeno + " byl přidán.");
    }

    private static void zaznamenatDochazku() {
        System.out.print("Zadejte jméno studenta: ");
        String jmeno = scanner.nextLine();
        for (Student student : studenti) {
            if (student.getJmeno().equalsIgnoreCase(jmeno)) {
                System.out.print("Je student přítomen? (ano/ne): ");
                String odpoved = scanner.nextLine();
                student.zaznamDochazky(odpoved.equalsIgnoreCase("ano"));
                System.out.println("Docházka zaznamenána.");
                return;
            }
        }
        System.out.println("Student nebyl nalezen.");
    }

    private static void zobrazitDochazku() {
        for (Student student : studenti) {
            System.out.println("Student: " + student.getJmeno() + " | Docházka: " + student.getDochazka());
        }
    }
}